package.json

```diff
@@ -45,10 +45,13 @@
     "url": "https://github.com/mozilla/fathom.git"
   },
   "scripts": {
+    "add-geckodriver-to-path": "export PATH=$PATH:$PWD/node_modules/geckodriver/bin",
     "coveralls": "cat ./coverage/lcov.info | coveralls",
     "lint": "make lint",
+    "init-server": "./test/functional/start_http_server & echo $! > pid",
     "pretest": "npm run lint",
-    "test": "make coverage"
+    "test": "npm run add-geckodriver-to-path && npm run init-server && make coverage && npm run uninit-server",
```

Comment:
You can actually keep the shell environment cleaner by avoiding the `export`: if you say this instead, the shell variable change will be in scope only for this one line:

```
PATH=$PATH:$PWD/node_modules/geckodriver/bin npm run add-geckodriver-to-path && npm run init-server && make coverage && npm run uninit-server
```

The only trouble is that none of this shell cleverness will work under Windows, where @danielhertenstein lives. I can't really think of a way to make it work, either, without a lot of extra code. Let's leave it like this for now and see if it causes him trouble. After all, `make test` should still run all but the browser tests, and he does have the option of falling back to a WSL shell if he wants to run those too.

Reply:
> You can actually keep the shell environment cleaner by avoiding the `export`: if you say this instead, the shell variable change will be in scope only for this one line:
> ```
> PATH=$PATH:$PWD/node_modules/geckodriver/bin npm run add-geckodriver-to-path && npm run init-server && make coverage && npm run uninit-server
> ```

Did you mean to remove `npm run add-geckodriver-to-path`? This is what I have now that is working:

```diff
diff --git a/package.json b/package.json
index bb87939..fa2a73c 100644
--- a/package.json
+++ b/package.json
@@ -45,12 +45,11 @@
     "url": "https://github.com/mozilla/fathom.git"
   },
   "scripts": {
-    "add-geckodriver-to-path": "export PATH=$PATH:$PWD/node_modules/geckodriver/bin",
     "coveralls": "cat ./coverage/lcov.info | coveralls",
     "lint": "make lint",
     "init-server": "./test/browser/start_http_server & echo $! > pid",
     "pretest": "npm run lint",
-    "test": "npm run add-geckodriver-to-path && npm run init-server && make coverage && npm run uninit-server",
+    "test": "PATH=$PATH:$PWD/node_modules/geckodriver/bin && npm run init-server && make coverage && npm run uninit-server",
     "uninit-server": "kill $(cat pid) && rm pid"
   },
   "main": "index"
```

> The only trouble is that none of this shell cleverness will work under Windows, where @danielhertenstein lives. I can't really think of a way to make it work, either, without a lot of extra code. Let's leave it like this for now and see if it causes him trouble. After all, `make test` should still run all but the browser tests, and he does have the option of falling back to a WSL shell if he wants to run those too.

Ah that's a good point. Actually, I had previously added the `--recursive` option to the `make test` command, which means it would try to run the JS functional tests at `test/browser` as well. It breaks currently, since I'm not doing the requisite set-up and clean-up. I'll remove that option for now, so `make test` doesn't run the JS functional tests at all. I'll add a comment to note this distinction in the Makefile.